File: tasks/translation/output/Qwen/Qwen2.5-32B-Instruct-AWQ/codenet/cot/Java/Python/s380356166_commented.py
import sys

class Solver:
    def solve(self, testNumber, in_stream, out_stream):
        ans = 0
        K = int(next(in_stream))
        S = int(next(in_stream))
        wk = 0
        for x in range(K + 1):
            for y in range(K + 1):
                wk = S - x - y
                if 0 <= wk <= K:
                    ans += 1
        out_stream.write(str(ans) + '\n')

class MyInput:
    def __init__(self, input_stream):
        self.input_stream = input_stream

    def nextInt(self):
        return int(next(self.input_stream))

def main():
    in_stream = iter(sys.stdin.read().split())
    out_stream = sys.stdout
    solver = Solver()
    solver.solve(1, in_stream, out_stream)

if __name__ == "__main__":
    main()
# End of Code

File: tasks/translation/output/Qwen/Qwen2.5-32B-Instruct-AWQ/codenet/mad/Java/C++/s380356166_commented.cpp
#include <iostream>
#include <vector>
#include <string>

class MyInput {
private:
    std::istream& in;

public:
    MyInput(std::istream& is) : in(is) {}

    int nextInt() {
        int value;
        in >> value;
        return value;
    }
};

class Solver {
public:
    void solve(int testNumber, MyInput& in, std::ostream& out) {
        int K = in.nextInt();
        int S = in.nextInt();
        int ans = 0;

        for (int x = 0; x <= K; x++) {
            for (int y = 0; y <= K; y++) {
                int z = S - x - y;
                if (z >= 0 && z <= K) {
                    ans++;
                }
            }
        }

        out << ans << std::endl;
    }
};

int main() {
    MyInput in(std::cin);
    std::ostream& out = std::cout;
    Solver solver;
    solver.solve(1, in, out);
    return 0;
}
// End of Code
